stl/inc/chrono

```diff
@@ -2086,6 +2087,689 @@ namespace chrono {
+    }
+    template <class _Duration>
+    _NODISCARD bool operator<=(const sys_time<_Duration>& _Left, const leap_second& _Right) noexcept {
+        return !(_Right.date() < _Left);
+    }
+
+    template <class _Duration>
+    _NODISCARD bool operator>=(const leap_second& _Left, const sys_time<_Duration>& _Right) noexcept {
+        return !(_Left.date() < _Right);
+    }
+    template <class _Duration>
+    _NODISCARD bool operator>=(const sys_time<_Duration>& _Left, const leap_second& _Right) noexcept {
+        return !(_Left < _Right.date());
+    }
+
+#ifdef __cpp_lib_concepts
+    template <class _Duration>
+    requires three_way_comparable_with<sys_seconds, sys_time<_Duration>> constexpr auto operator<=>(
+        const leap_second& _Left, const sys_time<_Duration>& _Right) noexcept {
+        return _Left.date() <=> _Right;
+    }
+    constexpr inline strong_ordering operator<=>(const leap_second& _Left, const leap_second& _Right) noexcept {
+        return _Left.date() <=> _Right.date();
+    }
+#endif // __cpp_lib_concepts
+
+    inline pair<vector<leap_second>, bool> _Xtzdb_generate_leap_seconds(const size_t _Current_size) {
+        // Returns empty vector if no new leap seconds are found.
+        constexpr static leap_second _Pre_2018_leap_seconds[]{
+            {sys_seconds{seconds{78796800}}, true},
+            {sys_seconds{seconds{94694400}}, true},
+            {sys_seconds{seconds{126230400}}, true},
+            {sys_seconds{seconds{157766400}}, true},
+            {sys_seconds{seconds{189302400}}, true},
+            {sys_seconds{seconds{220924800}}, true},
+            {sys_seconds{seconds{252460800}}, true},
+            {sys_seconds{seconds{283996800}}, true},
+            {sys_seconds{seconds{315532800}}, true},
+            {sys_seconds{seconds{362793600}}, true},
+            {sys_seconds{seconds{394329600}}, true},
+            {sys_seconds{seconds{425865600}}, true},
+            {sys_seconds{seconds{489024000}}, true},
+            {sys_seconds{seconds{567993600}}, true},
+            {sys_seconds{seconds{631152000}}, true},
+            {sys_seconds{seconds{662688000}}, true},
+            {sys_seconds{seconds{709948800}}, true},
+            {sys_seconds{seconds{741484800}}, true},
+            {sys_seconds{seconds{773020800}}, true},
+            {sys_seconds{seconds{820454400}}, true},
+            {sys_seconds{seconds{867715200}}, true},
+            {sys_seconds{seconds{915148800}}, true},
+            {sys_seconds{seconds{1136073600}}, true},
+            {sys_seconds{seconds{1230768000}}, true},
+            {sys_seconds{seconds{1341100800}}, true},
+            {sys_seconds{seconds{1435708800}}, true},
+            {sys_seconds{seconds{1483228800}}, true},
+        };
+        constexpr size_t _Pre_2018_count = _STD size(_Pre_2018_leap_seconds);
+
+        size_t _New_reg_ls_size;
+        auto _Reg_ls_data = unique_ptr<_RegistryLeapSecondInfo[]>{__std_tzdb_get_reg_leap_seconds(

```

Comment:
We conventionally avoid the "Almost Always Auto" pattern of `auto variable = Type{initializer};`. Directly initializing `Type variable{initializer};` is less verbose, and makes it clearer that the type information is being introduced instead of copied from another source.